pyproject.toml

```diff
@@ -1,108 +1,77 @@
 readme = "README.md"
-homepage = "https://github.com/temporalio/sdk-python"
-repository = "https://github.com/temporalio/sdk-python"
-documentation = "https://docs.temporal.io/docs/python"
-keywords = ["temporal", "workflow"]
-# We need to include proto source that is otherwise excluded via .gitignore.
-# We have chosen to keep all source including Rust source in precompiled wheels
-# for easy viewing. It is also complicated to exclude certain pieces for wheels
-# with Poetry (see https://github.com/python-poetry/poetry/issues/3380).
-include = ["temporalio/api/**/*", "temporalio/bridge/proto/**/*"]
-exclude = ["temporalio/bridge/**/target"]
-
-# Known undocumented API for hooking into setup. Unfortunately Poetry does not
-# support this script in a subdirectory like scripts/.
-[tool.poetry.build]
-script = "build.py"
-# In https://github.com/python-poetry/poetry-core/pull/318, Poetry stopped using
-# a setup file, but we are using setuptools Rust manually in our build.py which
-# needs a setup file.
-# TODO(cretz): Find a way to not require a setup file since that is going away
-# at some point in Poetry. Revisit Maturin or find some other approach.
-generate-setup-file = true
-
-[tool.poetry.urls]
+license = { file = "LICENSE" }
+keywords = [
+    "temporal",
+    "workflow",
+]
+dependencies = [
+    "protobuf>=3.20",
+    "python-dateutil>=2.8.2,<3 ; python_version < '3.11'",
+    "types-protobuf>=3.20",
+    "typing-extensions>=4.2.0,<5",
+]
+
+[project.optional-dependencies]
+grpc = ["grpcio>=1.48.2,<2"]
+opentelemetry = [
+    "opentelemetry-api>=1.11.1,<2",
+    "opentelemetry-sdk>=1.11.1,<2",
+]
+pydantic = ["pydantic>=2.0.0,<3"]
+
+[project.urls]
+Homepage = "https://github.com/temporalio/sdk-python"
+Repository = "https://github.com/temporalio/sdk-python"
+Documentation = "https://docs.temporal.io/docs/python"
 "Bug Tracker" = "https://github.com/temporalio/sdk-python/issues"
 
-[tool.poetry.dependencies]
-grpcio = {version = "^1.48.2", optional = true}
-opentelemetry-api = { version = "^1.11.1", optional = true }
-opentelemetry-sdk = { version = "^1.11.1", optional = true }
-protobuf = ">=3.20"
-pydantic = { version = "^2.0.0", optional = true }
-python = "^3.9"
-python-dateutil = { version = "^2.8.2", python = "<3.11" }
-types-protobuf = ">=3.20"
-typing-extensions = "^4.2.0"
-
-[tool.poetry.dev-dependencies]
-cibuildwheel = "^2.22.0"
-grpcio-tools = "^1.48.2"
-mypy = "^1.0.0"
-mypy-protobuf = "^3.3.0"
-psutil = "^5.9.3"
-pydocstyle = "^6.3.0"
-pydoctor = "^24.11.1"
-pyright = ">=1.1.377"
-pytest = "^7.4"
-pytest-asyncio = "^0.21"
-pytest-timeout = "^2.2"
-ruff = "^0.5.0"
-setuptools = ">=65.0.0"
-setuptools-rust = ">=1.3.0"
-toml = "^0.10.2"
-twine = "^4.0.1"
-wheel = "^0.42.0"
-
-[tool.poetry.extras]
-grpc = ["grpcio"]
-opentelemetry = ["opentelemetry-api", "opentelemetry-sdk"]
-pydantic = ["pydantic"]
-
-[tool.poetry.group.dev.dependencies]
-ruff = "^0.5.0"
+[dependency-groups]
+dev = [
+    "cibuildwheel>=2.22.0,<3",
```

Comment:
Do we need `cibuildwheel` anymore or can we use `maturin` to build the wheels?

Reply:
This is an area I'm unsure about also. I should have left a TODO. `uv build` builds the wheel in `dist/`, and it invokes `maturin` to do so (invoking `maturin build` directly builds the wheel in `temporalio/bridge`).

I am unsure exactly how `cibuildwheel` works. It must know how to build the `rust` extension; I _think_ that means it refers to the `[build-system]` in pyproject.toml, which leads it to invoke `maturin`? So, a bit of reasearch needed and also obviously need to verify the wheels built in CI are functioning correctly.

Reply:
`cibuildwheel` is a fairly common approach to building multiplatform wheels in docker images in CI. Yes, it does a traditional Python build I guess with the build system. But `maturin` says they offer most of this (ref https://www.maturin.rs/distribution.html#build-wheels and https://github.com/PyO3/maturin-action), but I haven't tested.

Reply:
Yes, maybe I'll do it incrementally. I'll probably move in the `maturin` direction. This GitHub workflow from `pydantic-core` is probably a reasonable model to consult: https://github.com/pydantic/pydantic-core/blob/main/.github/workflows/ci.yml#L480-L487